core/charm/arches.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"sort"
+)
+
+// Arch represents a platform architecture.
+type Arch string
+
+func (a Arch) String() string {
+	return string(a)
+}
+
+const (
+	// ArchAll represents an architecture for all architectures.
+	ArchAll Arch = "all"
+
+	// ArchAMD64 defines a amd64 architecture.
+	ArchAMD64 Arch = "amd64"
+
+	// ArchARM64 defines a arm64 architecture.
+	ArchARM64 Arch = "arm64"
+
+	// ArchPPC64 defines a ppc64 architecture.
+	ArchPPC64 Arch = "ppc64"
+
+	// ArchS390 defines a s390 architecture.
+	ArchS390 Arch = "s390"
+)
+
+// Arches defines a list of arches to compare against.
+type Arches struct {
+	set map[Arch]struct{}
+}
+
+// DefaultArches creates a series of arches to compare against.
+func DefaultArches() Arches {
+	return Arches{
+		set: map[Arch]struct{}{
+			ArchAll:   {},
+			ArchAMD64: {},
+			ArchARM64: {},
+			ArchPPC64: {},
+			ArchS390:  {},
+		},
+	}
+}
+
+// Contains checks to see if a given arch is found with in the list.
+func (a Arches) Contains(arch Arch) bool {
+	_, ok := a.set[arch]
+	return ok
+}
+
+// StringList returns an ordered list of strings.
+// ArchAll will always be the front of the slice to show importance of the enum
+// value.
+func (a Arches) StringList() []string {
```

Comment:
If you drop ArchAll and switch to a string set, you can change this to `String` to implement `fmt.Stringer` and have it simply return `strings.Join(set.SortedValues(), ",")

Reply:
Yeah, that was the reason for not using set.Strings.